src/Compilers/Core/Portable/MetadataReader/PEAssembly.cs

```diff
@@ -169,6 +169,14 @@ internal IEnumerable<ImmutableArray<byte>> GetInternalsVisibleToPublicKeys(strin
             return result ?? SpecializedCollections.EmptyEnumerable<ImmutableArray<byte>>();
         }
 
+        internal IEnumerable<string> GetInternalsVisibleToAssemblyNames()
+        {
+            if (_lazyInternalsVisibleToMap == null)
+                Interlocked.CompareExchange(ref _lazyInternalsVisibleToMap, BuildInternalsVisibleToMap(), null);
```

Comment:
Consider extracting this to a common method `EnsureInternalsVisibleToMapInitialized` or some such.